### DISCUSSION TITLE
ILSpy => myArray.Single where CSharpDecompiler.DecompileWholeModuleAsString => Enumerable.Single

### DISCUSSION THREAD

**Discussion by placebodk** [icsharpcode/ILSpy#3162] [Help] [answered]
Let me begin with praising this fantastic API which I believe have multiple use-cases. I am hoping someone on this forum can provide insight into the following:

When I use **ILSpy** to decompile a particular .NET Standard 2.0 assembly, this is the output of the decompile logic:

```
public static void SetEnable(string myString, MyType[] myTypes, out bool enable)
{
    enable = myTypes.Single((MyType s) => s.Name.Equals("myString)).Enable;
}
```

However, when I use the **ICSharpCode.Decompiler** directly in a C# .NET8 console program I get the following:

```
public static void SetEnable(string myString, MyType[] myTypes, out bool enable)
{
	enable = Enumerable.Single(myTypes, (MyType s) => s.Name.Equals("myString)).Enable;
}
```
Notice that the `myTypes.Single(...)` is decompiled into `Enumerable.Single(myTypes,...)` . While I am aware the code is technically the same, the goal is to have the output aligned with the ILSpy ouyput.

This is the code used to decompile from the console program:

```
var assemblyResolver = new UniversalAssemblyResolver(assemblyName, throwOnError: true, ".NETSTANDARD", "Microsoft.WindowsDesktop.App", PEStreamOptions.Default);
var decompiler = new CSharpDecompiler(assemblyName, assemblyResolver, new DecompilerSettings(LanguageVersion.Latest));
string code = decompiler.DecompileWholeModuleAsString();
```

Can anyone bring insight and hints which can help me align the output of the Console program to the output of ILSpy?

**Comment by siegfriedpammer** [comment, marked as answer]
The problem here is that the target framework parameter of the assembly resolver needs to consist of a framework name and version, just as TargetFrameworkAttribute specifies. In your case that would be ".NETStandard,Version=2.0".
